ferret/explainers/shap.py

```diff
@@ -49,15 +51,24 @@ def func(texts: np.array):
 
         masker = TextMasker(self.tokenizer)
         explainer_partition = shap.Explainer(model=func, masker=masker, **self.init_args)
-        shap_values = explainer_partition(text, **kwargs)
+        full_text = text
+        if isinstance(text, list):
```

Comment:
why do we need this double-check here? Is it some logic to handle another type of inference (e.g., the one for MultipleChoice tasks)? If that's the case, I would strongly advise against it and try to maintain the method as much general as possible (and let other classes provide the correct input).

Reply:
This check was indeed not needed. I now changed the `SequenceClassificationHelper._check_sample` and `ZeroShotTextClassificationHelper._prepare_sample` methods in the modeling/text_helpers.py files to handle properly the input type of both the NLI task and the Zero-Shot classification task. 
I have added extensive comments in the code to describe the procedure. Essentially, in the case in which the input is a tuple, e.g. `sample=(premise, hypothesis)` we concatenate the two elements of the input tuple via a `[SEP]` token, as this is consistent with the internal workings of the SequenceClassificationHelper.